### ISSUE TITLE
json_of(metadata) is broken in v2

### ISSUE THREAD

**Issue by martinkirch** [savonet/liquidsoap#1590]
After switching to 2.0.0-alpha1+dev Liquidsoap is not POSTing tracks' metadata correctly.  I'm using the following snippet :

```ocaml
def post_to_showergel(metadata)
  response = http.post(
    headers=[("Content-Type", "application/json")],
    "#{SHOWERGEL}/metadata_log",
    data=json_of(metadata)
  )
  log(response)
end
source.on_track(grille_avec_jingle, post_to_showergel)
```

so far I noticed that
 * Showergel receives `[  ]` when starting the stream (I guess that's the only sample ever sent by `blank()` at the bottom of my fallback)
 * unicode is not escaped correctly. For example `"©"` should be `"\u00a9"` but Liquidsoap sends `"\194\169"`
 * it seems that pictures embedded in metadata are encoded too! That yields big fields like `"apic": "\\000image/jpeg\\000\\003\\000\\255\\216\\255\\224\...`, so big the request sometimes gets rejected by `413 Payload Too Large`

**Comment by smimram** [comment]
Hi, I have just commited a patch which should solve the problem. `"©"` will stay the same, but this is apparently allowed. Can you confirm that it helps?

**Comment by dhannyz** [comment]
You may also want to use `map_metadata` (perhaps with update=false) earlier in your script and be explicit with what metadata you want to be POSTing as JSON

**Comment by martinkirch** [comment]
@smimram Yes it helps !

and clearly the initial `blank()` is still posting `[ ]` - it should be a JSON object with at least `on_air` and `source`.

about images, now it's crashing on:
```
2021/04/21 22:06:59 [clock.alsa:2] Source alsa failed while streaming: Failure("Utils.utf_8.next")!
```
Digging in older logs shows this already happened with 1.4.x : I bet it's #214 because all crashing tracks do have an embedded cover. I'm not sure what's the cleanest way to solve it... Would it be possible to exclude the metadata field if it's not proper unicode ?

**Comment by smimram** [comment]
You analysis is right: cover-art is not proper unicode data, and there is no easy/standard way of serializing raw data into JSON.

The easy way for now is to do something like
```
metadata = list.assoc.remove.all("cover", metadata)
```
to filter out cover from metadata.

**Comment by smimram** [comment]
After second thought, I have just commited code which silently tries to recover in invalid utf8 data rather than badly crashing.

Can we now close the bug? :)

**Comment by martinkirch** [comment]
After your commit I needed
```
  metadata = list.assoc.remove.all("cover", metadata)
  metadata = list.assoc.remove.all("apic", metadata)
```
to fix the POST - I can live with that.

I also noticed
![Capture d’écran du 2021-04-22 22-13-56](https://user-images.githubusercontent.com/324033/115779438-490f9500-a3b8-11eb-9296-be26d7ab0bb8.png)
(as the scollbar tells, `apic` is huge)

On this side I can easily add a finer UTF-decoder around telnet commands, no big deal.

I've noticed that v2.0 mostly (always?) decode with ffmpeg, whereas 1.4 mostly used MAD, maybe the difference just comes from that :thinking: especially as in 1.4, crashes happen on .m4a files for a similar reason :thinking: :thinking:  so your commit would improve their case, too.

Thanks @smimram  !
